parteh/PRTGenericMod.F90

```diff
@@ -0,0 +1,1196 @@
+
+  ! -------------------------------------------------------------------------------------
+  ! This type simply packs the names and symbols associated with all
+  ! the variables for any given hypothesis
+  ! -------------------------------------------------------------------------------------
+  
+  type :: state_descriptor_type
+     character(len=maxlen_varname)   :: longname
+     character(len=maxlen_varsymbol) :: symbol
+     integer                         :: organ_id    ! global id for organ
+     integer                         :: spec_id     ! global id for species
+     integer                         :: num_pos     ! number of descrete spatial positions
+
+     ! Also, will probably need flags to define different types of groups that this variable
+     ! belongs too, which will control things like fusion, normalization, when to zero, etc...
+
+  end type state_descriptor_type
+  
+
+
+  ! This type will help us loop through all the different variables associated
+  ! with a specific organ type. Since variables are a combination of organ and
+  ! species, the number of unique variables is capped at the number of species
+  ! per each organ.
+  
+  type organ_map_type
+     integer, dimension(1:num_species_types) :: var_id
+     integer                                 :: num_vars
+  end type organ_map_type
+
+
+  ! This structure packs both the mapping structure and the variable descriptors
+  ! -------------------------------------------------------------------------------------
+  ! This array should contain the lists of indices to 
+  ! the species x organ variable structure that is used to map variables to the outside
+  ! world.
+  !   
+  !              
+  !                   | carbon | nitrogen | phosphorous | .... |
+  !                   ------------------------------------------
+  !    leaf           |        |          |             |      |
+  !    fine-root      |        |          |             |      |
+  !    sapwood        |        |          |             |      |
+  !    storage        |        |          |             |      |
+  !    reproduction   |        |          |             |      |
+  !    structure      |        |          |             |      |
+  !    ....           |        |          |             |      |
+  !                   ------------------------------------------
+  !     
+  ! -------------------------------------------------------------------------------------
+
+
+  type prt_instance_type
+     
+     ! Note that index 0 is reserved for "all" or "irrelevant"
+     character(len=maxlen_varname)                             :: hyp_name
+
+     ! This will save the specific variable id associated with each organ and species
+     integer, dimension(0:num_organ_types,0:num_species_types) :: sp_organ_map
+
+     
+     type(state_descriptor_type), allocatable                  :: state_descriptor(:)
+
+     ! This will save the list of variable ids associated with each organ. There
```

Comment:
I keep getting tripped up by species, because I'm an ecologist type person. Is there a synonym (elements?) one might use?

Reply:
agree that this is confusing.  possible synonyms:  element, ingredient, nutrient, material, mineral, compound, chemical, nuclide, isotope, atom?